### PR TITLE
fix: Convert the update dialog to be a ContainedDialog

### PR DESCRIPTION
**Pull request by jalkire**
#### Details
Currently, our update dialog is a traditional modal dialog in a separate window. This results in some undesirable side effects while the dialog is open, such as the inability to reposition or close the application. To improve this behavior, this PR switches the update dialog to be a `ContainedDialog`, following the same pattern as our telemetry and channel switching dialogs. With one exception, the updated `UpdateContainedDialog` provides exactly the same functionality to the user as the existing `UpdateDialog`. The exception: for required updates, there is now a Close button in place of the Later button for option updates.

Screenshots capturing various color modes and combinations of stacked dialogs:
Mode | Screenshot
--- | ---
Light | image
Dark| image
HC 1 | image
HC 2 | image
HC Black | image
HC White | image

##### Motivation
Addresses issue #1186 (at least partially) 

##### Context
- To support this change, I updated `DialogContainerModeControl` to disable all background `ContainedDialog`. This way, in the (rare) scenario where the user has the Update, Telemetry, and StartUp dialogs all open at once, only the topmost contained dialog will show up in the tab order.

#### Pull request checklist

- [n/a] Run through of all [test scenarios](https://github.com/Microsoft/accessibility-insights-windows/blob/main/docs/Scenarios.md) completed?
- [x] Does this address an existing issue? If yes, Issue# - #1186
- [x] Includes UI changes?
  - [x] Run the production version of Accessibility Insights for Windows against a version with changes.
  - [x] Attach any screenshots / GIF's that are applicable.

> Note: After the PR has been created, certain checks will be kicked off. All of these checks must pass before a merge.

### CODE REVIEW

File: src/AccessibilityInsights/Modes/DialogContainerModeControl.xaml.cs [hunk 1]
```diff
@@ -39,12 +50,22 @@ protected override AutomationPeer OnCreateAutomationPeer()
             if (gdContainer.Children.Count == 0)
             {
                 this.Visibility = Visibility.Collapsed;
+            } else
```
Comment (DaveTryon) [inline src/AccessibilityInsights/Modes/DialogContainerModeControl.xaml.cs:53]: "else" on separate line

File: src/AccessibilityInsights.SharedUx/Properties/Resources.resx [hunk 2]
```diff
@@ -1515,4 +1515,31 @@ Do you want to change the channel? </value>
   <data name="TextRangeControl_Details" xml:space="preserve">
     <value>Details</value>
   </data>
+  <data name="UpdateDialogWindowTitle" xml:space="preserve">
+    <value>There is an update available</value>
```
Comment (DaveTryon) [inline src/AccessibilityInsights.SharedUx/Properties/Resources.resx:1519]: `UpdateDialogWindowTitle` appears to be unused

**Review by DaveTryon — APPROVED** [review]
Approved with a couple of comments

File: src/AccessibilityInsights.SharedUx/Properties/Resources.resx (hunk 2 above)
Comment (jalkire) [reply src/AccessibilityInsights.SharedUx/Properties/Resources.resx:1519]: This is used on line 12 of `UpdateContainedDialog`. It's still got the old name referring to a window, so I have now updated that to be more accurate

**Pull request merged**